src/Microsoft.Health.Fhir.Core/Features/Operations/Export/ExportJobTask.cs

```diff
@@ -390,5 +409,58 @@ private async Task ProcessSearchResultsAsync(IEnumerable<SearchResultEntry> sear
                 await UpdateJobRecordAsync(cancellationToken);
             }
         }
+
+        private async Task<SearchResult> GetGroupPatients(string groupId, List<Tuple<string, string>> queryParametersList, DateTimeOffset groupMembershipTime, CancellationToken cancellationToken)
+        {
+            if (!queryParametersList.Exists((Tuple<string, string> parameter) => parameter.Item1 == KnownQueryParameterNames.Id))
+            {
+                var patientIds = await GetGroupPatientIds(groupId, groupMembershipTime, cancellationToken);
+                queryParametersList.Add(Tuple.Create(KnownQueryParameterNames.Id, string.Join(',', patientIds)));
+            }
+
+            using (IScoped<ISearchService> searchService = _searchServiceFactory())
+            {
+                return await searchService.Value.SearchAsync(
+                               resourceType: null,
+                               queryParametersList,
+                               cancellationToken);
+            }
+        }
+
+        private async Task<HashSet<string>> GetGroupPatientIds(string groupId, DateTimeOffset groupMembershipTime, CancellationToken cancellationToken, HashSet<string> groupsAlreadyChecked = null)
+        {
+            if (groupsAlreadyChecked == null)
+            {
+                groupsAlreadyChecked = new HashSet<string>();
+            }
+
+            groupsAlreadyChecked.Add(groupId);
+
+            var groupContents = await _groupMemberExtractor.GetGroupMembers(groupId, groupMembershipTime, cancellationToken);
+
+            var patientIds = new HashSet<string>();
+
+            foreach (Tuple<string, string> entity in groupContents)
+            {
+                var resourceId = entity.Item1;
+                var resourceType = entity.Item2;
+                switch (resourceType)
+                {
+                    case KnownResourceTypes.Patient:
```

Comment:
nit: we should add a comment here explaining why we consider only Patient and Group type and skip the resource types.